src/System.Diagnostics.Tracing/tests/BasicEventSourceTest/TestsManifestGeneration.cs

```diff
@@ -0,0 +1,179 @@
+                    ValidateManifest(baselineManFile, manfilename);
+                }
+                TestUtilities.CheckNoEventSourcesRunning("Stop");
+            }
+#endif
+        }
+
+        /// <summary>
+        /// Calls GenerateManifest() on the event source passed in, saves the output to a file under 'folder'
+        /// and returns the filename under which the manifest was saved.
+        /// </summary>
+        /// <param name="eventSourceType">The EventSource derived class whose manifest we want saved</param>
+        /// <param name="folder">The folder where the manifest will be saved</param>
+        /// <returns>The filename under 'folder' that contains the ETW manifest</returns>
+        private string SaveEventSourceManifest(Type eventSourceType, string folder)
+        {
+            var manfilename = Path.Combine(folder, GetPrefixFromType(eventSourceType) + eventSourceType.Name + ".man");
+            using (var manfile = new System.IO.StreamWriter(new System.IO.MemoryStream(Encoding.Unicode.GetBytes(manfilename))))
+            {
+                string man = null;
+                string dllName = Path.GetFileName(eventSourceType.GetTypeInfo().Assembly.Location);
+                //if (!eventSourceType.GetTypeInfo().Assembly.ReflectionOnly)
+                //{
+                    var baseAssm = eventSourceType.GetTypeInfo().BaseType.GetTypeInfo().Assembly;
+                    var tyGmf = (baseAssm != null) ? baseAssm.GetType(eventSourceType.GetTypeInfo().BaseType.Namespace + ".EventManifestOptions") : null;
+                    MethodInfo mi = null;
+                    if (tyGmf != null)
+                    {
+                        //mi = eventSourceType.GetTypeInfo().GetMethod("GenerateManifest", BindingFlags.Static | BindingFlags.FlattenHierarchy | BindingFlags.Public,
+                        //                               null, new Type[] { typeof(Type), typeof(string), tyGmf }, null);
+                        mi = eventSourceType.GetTypeInfo().GetDeclaredMethod("GenerateManifest");
+                    }
+                    if (mi != null)
+                    {
+                        man = (string)mi.Invoke(null, new object[] { eventSourceType, dllName, 2 });
+                    }
+                    //else
+                    //{
+                    //    mi = eventSourceType.GetTypeInfo().GetMethod("GenerateManifest", BindingFlags.Static | BindingFlags.FlattenHierarchy | BindingFlags.Public,
+                    //                                        null, new Type[] { typeof(Type), typeof(string) }, null);
+                    //    man = (string)mi.Invoke(null, new object[] { eventSourceType, dllName });
+                    //}
+                //}
+                //else
+                //{
+                //    man = EventSource.GenerateManifest(eventSourceType, dllName, EventManifestOptions.AllCultures);
+                //}
+                manfile.Write(man);
```

Comment:
Commented out code like this should either be removed or there should be a TODO with a GitHub issue number associated with it for when it will be uncommented.